### DISCUSSION TITLE
大家都在用哪种好的方法来解决前后端接口文档对接

### DISCUSSION THREAD

**Discussion by xmsz** [midwayjs/midway#1238] [Q&A]
背景
- 我们微服务用Protobuf定义
- BFF用Midway通过gRPC调用，这个流程通信很好，没什么问题
- 然后就是BFF => 前端了，这一步没想到怎么搞才是最好
- 然后我们部门很小也没有特殊需求，效率最重要

大家都是怎么解决的？

-----------------------
前端需要的什么
1. 请求的信息（地址、参数类型、返回类型）
2. 直接调用（一种是调用拿到配置，一种是直接封装request）

-----------

想象中的情况：
1. 前端和后端共有一个文件就能解决，类似于proto文件或者生成typescript文件更好
2. 类似与midway的hooks方式，但是这个要依赖前端框架

-----------------------
曾经的尝试

1. 手写文档，这个肯定是最累最麻烦
2. 生成SwaggerUI，这个只是解决了自动生成文档，但是没有能解决直接调用的需求，因为没有能给前端代码产物
3. 生成SwaggerJSON，这个稍微封装一下也是能用的，但还是感觉不够智能
4. 还有一些swagger2ts的工具

**Comment by waitingsong** [comment]
> 前端和后端共有一个文件就能解决，类似于proto文件或者生成typescript文件更好

可以考虑单独发布 NPM 接口包（可以和主服务在同一个 git 仓库中，使用 lerna 管理），接口包使用 ts 编写，前后端都可以使用安装依赖方式来获得接口信息（包括备注）。
这种方式可以解决接口数据共享问题，标准文档生成需要想办法。
